test/convenience/keyboard.test.ts

```diff
@@ -63,6 +72,83 @@ describe("Keyboard", () => {
         assertEquals(keyboard.resize_keyboard, false);
         assertEquals(keyboard.input_field_placeholder, "placeholder");
     });
+
+    it("can be transposed", () => {
+        function t(btns: string[][], expected: string[][]) {
+            assertEquals(
+                Keyboard.from(btns).toTransposed(),
+                Keyboard.from(expected),
+            );
+        }
+        t([["a"]], [["a"]]);
+        t([["a", "b", "c"]], [["a"], ["b"], ["c"]]);
+        t([["a", "b"], ["c", "d"], ["e"]], [["a", "c", "e"], ["b", "d"]]);
+        t(
+            [["a", "b"], ["c"], ["d", "e", "f"]],
+            [["a", "c", "d"], ["b", "e"], ["f"]],
+        );
+        const keyboard = Keyboard.from([["a", "b", "c"], ["d", "e"], ["f"]]);
+        assertEquals(keyboard.toTransposed().toTransposed(), keyboard);
```

Comment:
This is begging for a property-based test.

Reply:
We don't do those yet. We can consider adding new test setups in future PQs.